### PR TITLE
Improve worker-pool example

### PR DESCRIPTION
**Pull request by kipanshi**
Added collecting the fetched data together for further use,
refactored queuing,
Added some comments.

### CODE REVIEW

File: examples/worker-pool.php [hunk 1]
```diff
@@ -2,14 +2,26 @@
 <?php
 require dirname(__DIR__).'/vendor/autoload.php';
 
+
```
Comment (kelunik) [inline examples/worker-pool.php]: Please remove the unnecessary white space change. Maybe add a newline before the `require`.

File: examples/worker-pool.php [hunk 2]
```diff
@@ -44,3 +48,8 @@
 
     return yield $pool->shutdown();
 });
+
+assert(3 === count(array_keys($results)));
```
Comment (kelunik) [inline examples/worker-pool.php]: What's the reason for the `assert`?

Comment (kipanshi) [reply examples/worker-pool.php]: Removed.

File: examples/worker-pool.php (hunk 1 above)
Comment (kipanshi) [reply examples/worker-pool.php]: Fixed.

**Comment by kelunik** [comment]
Thanks!

**Pull request merged**